### PR TITLE
FALCON-2260 Enhance extension List api to support for user extensions

### PR DESCRIPTION
**Pull request by sandeepSamudrala**

### CODE REVIEW

File: webapp/src/test/java/org/apache/falcon/resource/ExtensionManagerIT.java [hunk 1]
```diff
@@ -100,8 +100,6 @@ public void testTrustedExtensionJob() throws Exception {
         // validate job list results
         ExtensionJobList jobs = context.getExtensionJobs("hdfs-mirroring", null, null, null, null, null);
         Assert.assertEquals(jobs.numJobs, 2);
-        Assert.assertEquals(jobs.job.get(0).jobName, JOB_NAME_1);
```
Comment (PraveenAdlakha) [inline webapp/src/test/java/org/apache/falcon/resource/ExtensionManagerIT.java]: Can we unable this test case and see if test cases are running correctly.Right now the test cases have been disabled.

@Test (enabled = false)

File: extensions/src/main/java/org/apache/falcon/extensions/jdbc/ExtensionMetaStore.java [hunk 2]
```diff
@@ -132,6 +132,18 @@ public ExtensionBean getDetail(String extensionName) {
         }
     }
 
+    public List<ExtensionJobsBean> getJobsForAnExtension(String extensionName) {
```
Comment (PraveenAdlakha) [inline extensions/src/main/java/org/apache/falcon/extensions/jdbc/ExtensionMetaStore.java]: Please add unit test for this.

File: common/src/main/java/org/apache/falcon/persistence/PersistenceConstants.java [hunk 3]
```diff
@@ -80,5 +80,6 @@ private PersistenceConstants(){
     public static final String GET_ALL_EXTENSION_JOBS = "GET_ALL_EXTENSION_JOBS";
     public static final String DELETE_EXTENSION_JOB = "DELETE_EXTENSION_JOB";
     public static final String GET_EXTENSION_JOB = "GET_EXTENSION_JOB";
+    public static final String GET_JOBS_FOR_AN_EXTENSION = "GET_JOBS_FOR_AN_EXTENSION";
```
Comment (sandeepSamudrala) [inline common/src/main/java/org/apache/falcon/persistence/PersistenceConstants.java]: Was thinking the same initially, but then to get all details I can always create a new query. I have made the changes accordingly.

File: extensions/src/main/java/org/apache/falcon/extensions/jdbc/ExtensionMetaStore.java (hunk 2 above)
Comment (sandeepSamudrala) [inline extensions/src/main/java/org/apache/falcon/extensions/jdbc/ExtensionMetaStore.java]: Ack. Added.

File: webapp/src/test/java/org/apache/falcon/resource/ExtensionManagerIT.java (hunk 1 above)
Comment (sandeepSamudrala) [inline webapp/src/test/java/org/apache/falcon/resource/ExtensionManagerIT.java]: This Test still doesn't seem to be working.

**Comment by pallavi-rao** [comment]
I have no further comments. Regarding, ExtensionManagerIT, I think it would be a good idea to park a JIRA to ensure trusted extension is not regressed.

**Comment by pallavi-rao** [comment]
Just one comment, the description of the list option needs to be changed:
Currently it is: `-list                   List extension jobs and associated entities`

**Comment by sandeepSamudrala** [comment]
@pallavi-rao : I had modified the documentation accordingly and I created the following jira to address the test case.
https://issues.apache.org/jira/browse/FALCON-2262

**Comment by pallavi-rao** [comment]
Thanks @sandeepSamudrala. @PraveenAdlakha, please re-review.

**Comment by PraveenAdlakha** [comment]
+1

**Pull request closed without merging**